action/protocol/staking/handlers.go

```diff
@@ -551,18 +583,30 @@ func (p *Protocol) fetchBucket(
 	index uint64,
 	checkOwner bool,
 	allowSelfStaking bool,
-) (*VoteBucket, error) {
+) (*VoteBucket, *fetchError) {
 	actionCtx := protocol.MustGetActionCtx(ctx)
 	bucket, err := getBucket(sr, index)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to fetch bucket by index %d", index)
+		fetchErr := &fetchError{
+			err:           errors.Wrapf(err, "failed to fetch bucket by index %d", index),
+			failureStatus: iotextypes.ReceiptStatus_ErrInvalidBucketIndex,
+		}
+		return nil, fetchErr
 	}
 	if checkOwner && !address.Equal(bucket.Owner, actionCtx.Caller) {
-		return nil, fmt.Errorf("bucket owner does not match action caller, bucket owner %s, action caller %s",
-			bucket.Owner.String(), actionCtx.Caller.String())
+		fetchErr := &fetchError{
+			err: fmt.Errorf("bucket owner does not match action caller, bucket owner %s, action caller %s",
+				bucket.Owner.String(), actionCtx.Caller.String()),
+			failureStatus: iotextypes.ReceiptStatus_ErrUnauthorizedOperator,
```

Comment:
this error falls in first type, that is we charge gas, and let Mint/Commit() continue